### PR TITLE
Different day? Different map rotation rules.

### PR DESCRIPTION
**Pull request by AffectedArc07**
## What Does This PR Do
The ending of the box monotony has now come to this. 

HERE ARE THE RULES
- Tuesdays and Thursdays will have fully randomised maps
- Weekends will have it so the map vote will not include the current map
- Other days (Mon/Wed/Fri) are unaffected

## Why It's Good For The Game
The box monotony needs to end. Theres no point telling people to put effort into other maps if we dont get them used, likewise, people are starting to feel the burnout from us having other choices but them not being used.

The whole issue with box is that of people refusing to learn and adapt to new maps for a varied playstyle. While a case could be made for this years back when we tried to push some rotation in 2018, things such as the webmap didnt exist, and the maps were in an unplayable state, compared to now where we have a map linked ingame, and all maps are functional.

## Testing
- Started local server
- Tried today's combo (no current map in the vote list)
- The current map was not in the vote list

## Changelog
:cl: AffectedArc07
tweak: Some map rotation types now appear on some days 
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/maprotate.dm [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		log_startup_progress("Somehow, we failed to extract a valid numerical day from the DB. ?????????????")
+		return ..()
+
+	switch(day_index)
+		if(2, 4) // Tuesday or thursday
+			log_startup_progress("Its Tuesday or Thursday, which means there is no map vote, thus its randomised!")
+			rotation_mode = MAPROTATION_MODE_FULL_RANDOM
+
+		if(6, 7) // Saturday or sunday, weekend
+			log_startup_progress("Its a weekend, which means the map vote wont contain the current map")
```
Comment (Edan52) [inline code/controllers/subsystem/maprotate.dm:52]:
```suggestion
			log_startup_progress("It's a weekend, which means the map vote wont contain the current map")
```

**Comment by AffectedArc07** [comment]
This is no longer a data only PR, it has merge intent now

File: code/controllers/subsystem/maprotate.dm [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		if(4)
+			return "Thursday"
+		if(5)
+			return "Friday"
+		if(6)
+			return "Saturday"
+		if(7)
+			return "Sunday"
+		else
+			return "A day that does not exist in the Gregorian Calendar."
```
Comment (hal9000PR) [inline code/controllers/subsystem/maprotate.dm:88]: would it be nicer to do a assoc list then use the number as an index?

Comment (AffectedArc07) [reply code/controllers/subsystem/maprotate.dm:88]: That would help with readability but this executes faster given its all static and not list lookups. Up to you if you want readability or performance on something that runs once.

Comment (hal9000PR) [reply code/controllers/subsystem/maprotate.dm:88]: I think a list would look better, its a tiny difference in performance.

Comment (AffectedArc07) [reply code/controllers/subsystem/maprotate.dm]: Done

File: code/controllers/subsystem/maprotate.dm [hunk 3]
```diff
@@ -0,0 +1,78 @@
+SUBSYSTEM_DEF(maprotate)
+	name = "Maprotate"
+	flags = SS_NO_FIRE
+
+	var/rotation_mode = MAPROTATION_MODE_NORMAL_VOTE
+	var/setup_done = FALSE
+
+// Debugging purposes. Im not having people change this on the fly.
+/datum/controller/subsystem/maprotate/vv_edit_var(var_name, var_value)
+	if (((var_name == "rotation_mode") || (var_name == "setup_done")) && !check_rights(R_MAINTAINER))
```
Comment (hal9000PR) [inline code/controllers/subsystem/maprotate.dm:10]:
```suggestion
	if(((var_name == "rotation_mode") || (var_name == "setup_done")) && !check_rights(R_MAINTAINER))
```
one last thing

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**